twarc/client.py

```diff
@@ -143,8 +143,8 @@ def timeline(self, user_id=None, screen_name=None, max_id=None,
             for status in statuses:
                 # If you request an invalid user_id, you may still get
                 # results so need to check.
-                if not user_id or user_id == status.get("user",
-                                                        {}).get("id_str"):
+                if not user_id or id == status.get("user",
```

Comment:
I haven't really traced the logic, but was this a significant change from `user_id` to `id`?

Reply:
I just added the validation for screen_name/user_id arguments, since this PR is right there anyway :)